Add .codeclimate.yml for running locally

toddmilliken opened the pull request:
## Summary

This PR adds the `.codeclimate.yml` file which will serve two purposes:

1. It will be used instead of the In-App configuration added to the site; so GitHub will now use this file for analyzing code.
2. It will also be used for running codeclimate locally, so errors can fixed before pushing up to GitHub.

This file was initially taken from bu-banners/.codeclimate.yml.

Then edits were made after installing codeclimate on my machine and then running commands like `codeclimate validate-config` which noted to change `exclude_paths` to `exclude_patterns`, add a `version` key, etc.

Then the key under plugins > phpcodesniffer > config > standard needed to be changed from `phpcs.xml` to `phpcs.xml.dist` so it doesn’t error on phpcs engine.

This PR also adds files to .gitignore that are fetched in .codeclimate.yml's prepare > fetch key. These files aren't going to be modified so they should be ignored.

Instructions also exist in the README.

## Instructions for testing running codeclimate locally

It is easiest to install codeclimate on your machine via homebrew. http://brew.sh . Once homebrew is installed, follow the steps outlined on the GitHub website for codeclimate https://github.com/codeclimate/codeclimate/:

1. run `brew tap codeclimate/formulae`
2. run `brew install codeclimate`

Now the codeclimate command should be available for use in the root of the repo. Can be tested by checking out this branch and running something like `codeclimate analyze includes/publications.php` for a specific file or just `codeclimate analyze` for a complete repo analysis. Note the commands should be run from the root of the repo where the `.codeclimate.yml` file exists.

toddmilliken committed:
Add .codeclimate.yml for running locally

This commit adds the .codeclimate.yml file which will serve two purposes:

1. It will be used instead of the In-App configuration added to the site; so GitHub will now use this file for analyzing code.
2.It will also be used for running codeclimate locally, so errors can fixed before pushing up to GitHub.

This file was initially taken from bu-banners/.codeclimate.yml.

Then edits were made after installing codeclimate on my machine and then running commands like codeclimate validate-config which noted to change exclude_paths to exclude_patterns, add a version key.

Then the key under plugins > phpcodesniffer > config > standard needed to be changed from `phpcs.xml` to `phpcs.xml.dist` so it doesn’t error on phpcs engine.

It is easiest to install codeclimate on your machine via homebrew. http://brew.sh . Once homebrew is installed, follow the steps outlined on the GitHub website for codeclimate https://github.com/codeclimate/codeclimate/:

1. run `brew tap codeclimate/formulae`
2. run `brew install codeclimate`

Now the codeclimate command should be available for use in the root of the repo. Can be tested by running something like `codeclimate analyze includes/publications.php` for a specific file or just `codeclimate analyze` for a complete repo analysis.

This PR also adds files to .gitignore that are fetched in .codeclimate.yml's prepare > fetch key. These files aren't going to be modified so they should be ignored.

These instructions also exist in the README.

toddmilliken committed:
Update codeclimate instructions

Add note to install Docker or Docker for Mac. This is a required prerequisite to make this work.

anton-kachurin commented on README.md:
```diff
@@ -37,3 +37,62 @@ One potential fix for this specific scenario is to symlink the php default socke
+`codeclimate version`.
+
+A list of commands is available on the [Code Climate GitHub page](https://github.com/codeclimate/codeclimate/#commands).
+The following commands can be run in the root of your repository:o
```
```suggestion
The following commands can be run in the root of your repository:
```

anton-kachurin reviewed (approved):
Works great! Worth noting that the first run (even for a single file) takes a lot of time. I suspect that it installs those tools that run during analysis during the first run.

anton-kachurin committed:
Update README.md

Typo

Co-Authored-By: toddmilliken <milliktr@gmail.com>

toddmilliken committed:
Update README with Docker notes for code climate

Need to run the `docker pull codeclimate/codeclimate` command before doing anything else with homebrew.

toddmilliken committed:
Clarify code climate installation notes

toddmilliken committed:
Simplify headings
